pkg/cloudprovider/cloudprovider.go

```diff
@@ -168,3 +289,60 @@ func (c *CloudProvider) Name() string {
 func (c *CloudProvider) GetSupportedNodeClasses() []status.Object {
 	return []status.Object{&v1alpha1.GCENodeClass{}}
 }
+
+func (c *CloudProvider) instanceToNodeClaim(i *instance.Instance, instanceType *cloudprovider.InstanceType) *karpv1.NodeClaim {
+	nodeClaim := &karpv1.NodeClaim{}
+	labels := map[string]string{}
+	annotations := map[string]string{}
+
+	if instanceType != nil {
+		labels = utils.GetAllSingleValuedRequirementLabels(instanceType)
+
+		resourceFilter := func(name corev1.ResourceName, value resource.Quantity) bool {
+			return !resources.IsZero(value)
+		}
+
+		nodeClaim.Status.Capacity = lo.PickBy(instanceType.Capacity, resourceFilter)
+		nodeClaim.Status.Allocatable = lo.PickBy(instanceType.Allocatable(), resourceFilter)
+	}
+
+	// Set core labels
+	labels[corev1.LabelTopologyZone] = i.Location
+	labels[karpv1.CapacityTypeLabelKey] = i.CapacityType
+
+	// Add node pool label if present
+	if v, ok := i.Labels[karpv1.NodePoolLabelKey]; ok {
+		labels[karpv1.NodePoolLabelKey] = v
+	}
+
+	nodeClaim.Labels = labels
```

Comment:
Wouldn't it make sense to assign it directly to `nodeClaim` structure from `if instanceType != nil`

Reply:
Thank you for your suggestion, but when instance type is nil, we can still set the karpenter label for nodeclaim to facilitate subsequent processing by karpenter, so we can keep the current implementation.